stream_alert/athena_partition_refresh/main.py

```diff
@@ -0,0 +1,206 @@
+    if not query_results_resp['ResultSet']['Rows']:
+        LOGGER.debug('The query %s returned no data', kwargs['query'])
+
+    return query_results_resp
+
+
+def check_database_exists(results_bucket, results_path):
+    """Verify the StreamAlert Athena database exists."""
+    query_resp = run_athena_query(
+        query='SHOW DATABASES LIKE \'streamalert\';',
+        results_bucket=results_bucket,
+        results_path=results_path
+    )
+    if isinstance(query_resp, dict) and not query_resp['ResultSet']['Rows']:
+        LOGGER.info('The \'streamalert\' database does not exist, please create it.')
+        return False
+
+    return True
+
+
+def check_table_exists(results_bucket, results_path, table_name):
+    """Verify a given StreamAlert Athena table exists."""
+    query_resp = run_athena_query(
+        query='SHOW TABLES LIKE \'{}\';'.format(table_name),
+        database='streamalert',
+        results_bucket=results_bucket,
+        results_path=results_path
+    )
+    if isinstance(query_resp, dict) and not query_resp['ResultSet']['Rows']:
+        LOGGER.info('The streamalert table \'%s\' does not exist, please create it.', table_name)
+        return False
+
+    return True
+
+
+def normal_partition_refresh(config, athena_results_bucket, athena_results_path):
+    normal_partition_config = config['lambda']['athena_partition_refresh_config']['partitioning']['normal']
+    for _, athena_table in normal_partition_config.iteritems():
+        resp = run_athena_query(
+            query='MSCK REPAIR TABLE {};'.format(athena_table),
+            database='streamalert',
+            results_bucket=athena_results_bucket,
+            results_path=athena_results_path
+        )
+        if resp:
+            LOGGER.info('Query results:')
+            for row in resp['ResultSet']['Rows']:
+                LOGGER.info(row)
+
+
+def firehose_partition_refresh(_):
+    LOGGER.error('Firehose partition refresh is not yet supported, exiting!')
+    raise NotImplementedError
+
+
+def handler(event, context):
+    """Athena Partition Refresher Handler Function"""
+    config = _load_config()
+    if not config:
+        LOGGER.error('No config found, exiting!')
+        return
+
+    global ATHENA_CLIENT
+    ATHENA_CLIENT = boto3.client('athena', region_name=config['global']['account']['region'])
+
+    # Athena queries need an S3 bucket for results
+    athena_results_bucket = 's3://aws-athena-query-results-{}-{}'.format(
+        config['global']['account']['aws_account_id'],
+        config['global']['account']['region']
+    )
+    # Produces athena_partition_refresh/2017/01/01 keys
+    athena_results_path = 'athena_partition_refresh/{}'.format(
+        datetime.now().strftime('%Y/%m/%d'))
+
+    # The StreamAlert database needs to exist before we run queries.
+    if not check_database_exists(athena_results_bucket, athena_results_path):
+        return
```

Comment:
Another example of a critical error that should probably bubble all the way up instead of failing silently.